pkg/scheduler/scheduler.go

```diff
@@ -85,24 +85,42 @@ func NewScheduler(logC log.Logger) *Scheduler {
 	}
 }
 
-const rateLimit = 5 * time.Second
+const (
+	rateLimit  = 5 * time.Second
+	numWorkers = 5
+)
+
+func (scheduler *Scheduler) taskWorker(jobs chan Task) {
+	for j := range jobs {
```

Comment:
This for loop will run indefinitely until the channel jobs is closed?

Reply:
yes